### PR TITLE
Added has_many intervention_participations validation for product

### PR DESCRIPTION
**Pull request by Zank94**
Answer to #2057

### CODE REVIEW

**Comment by Zank94** [comment]
[ekylibre:debug/2057-Undefined-method-intervention-participations-in-equipments#show](https://trello.com/c/RJBKKscD/287-ekylibredebug-2057-undefined-method-intervention-participations-in-equipmentsshow)

File: test/models/product_test.rb [hunk 1]
```diff
@@ -94,4 +94,31 @@ class ProductTest < ActiveSupport::TestCase
       assert Product.of_working_set(item.name).count >= 0
     end
   end
+
+  test 'intervention_participations association' do
```
Comment (Aquaj) [inline test/models/product_test.rb]: Could you rename the test to be more descriptive? (something along the lines of "product has a way to get its intervention_participations" maybe ?)

File: test/models/product_test.rb [hunk 2]
```diff
@@ -94,4 +94,31 @@ class ProductTest < ActiveSupport::TestCase
...
+                category: category,
+                nature: nature,
+                variant: variant,
+                variety: "cultivable_zone"
+              )
+    intervention_participation = InterventionParticipation.create!(
+                                   product: product,
+                                   state: "in_progress"
+                                   )
+    assert_equal(1, product.intervention_participations.length)
```
Comment (Aquaj) [inline test/models/product_test.rb]: Could you remove the parens ?

Comment (Aquaj) [inline test/models/product_test.rb]: Could you check the `product.intervention_participations` collection does contain the `intervention_participation` you created before ?

**Pull request merged**